### DISCUSSION TITLE
Questions

### DISCUSSION THREAD

**Discussion by rcosta358** [cojen/Maker#17] [Q&A] [answered]
Hi there, I have two questions,

1. Is it possible to get the value of a Variable and how if so? For example:

`Variable paramVar = handleMaker.param(idx).invoke("get", p.getName());`
`String paramValue = ???`

2. Is there any way that I can call a function from the java standard library? 
For example, how do I call the Integer.parseInt function? Something like this:

`ClassMaker cm = ClassMaker.begin().classLoader().loadClass("java.lang.Integer");`
`MethodMaker mm = cm.getMethod("parseInt", String.class);`
`Variable intParamVar = mm.invoke("parseInt", paramVar);`

Thanks a lot.

**Comment by broneill** [comment, marked as answer]
1. I don't understand what you mean here. The parameter doesn't have a value except when the code is finished and it's run.
2. var intParamVar = mm.var(Integer.class).invoke("parseInt", paramVar);

**Comment by rcosta358** [comment]
1. So I can't get the value of the variable when the bytecode is run?
2. Ok that makes sense, should have thought about that, thanks.

**Comment by broneill** [comment]
In what form do you want the variable to be seen when the code is run? Just printed out? This will do that: `mm.var(System.class).field("out").invoke("println", paramVar);`

**Comment by rcosta358** [comment]
The first solution worked, so yeah I don't need to get the value itself to then call the parseInt function. Thank you.
